controller/konnect/konnectextension_controller_secrets.go

```diff
@@ -30,26 +32,43 @@ func listKonnectExtensionsBySecret(ctx context.Context, cl client.Client, s *cor
 		return nil, err
 	}
 
+	for _, ownerRef := range s.GetOwnerReferences() {
+		if ownerRef.Controller != nil && *ownerRef.Controller {
+			owner := &konnectv1alpha1.KonnectExtension{}
+			err := cl.Get(ctx, k8stypes.NamespacedName{
+				Namespace: s.Namespace,
+				Name:      ownerRef.Name,
+			}, owner)
+			if err != nil {
+				return nil, err
+			}
+			if owner.UID == ownerRef.UID {
+				l.Items = append(l.Items, *owner)
+			}
```

Comment:
🤔 Do we actually need to compare the UIDs here? Wouldn't it be possible to just use: `controllerutil.HasOwnerReference` (possibly with `controllerutil.HasControllerReference`? Although I see that the latter just returns a bool, taking in no arguments 🤔 )?

`HasOwnerReference` verifies owners' group, kind and name and I think that should be enough for this case, no?

Reply:
I have added the check to `Kind` and `APIVersion` and removed the UID check.